docs/src/main/asciidoc/getting-started.adoc

```diff
@@ -425,6 +425,21 @@ URL testUrl;
 The test port can be controlled via the `quarkus.http.test-port` config property. Quarkus also creates a system
 property called `test.url` that is set to the base test URL for situations where you cannot use injection.
 
+== Working with multi-module project or external modules
+
+Quarkus uses Jandex to create, at build time, an index of all classes eligible to CDI bean discovery.
+Without it, none of the Quarkus extensions will work.
+
+This index is created by default on your project thanks to our Maven or Gradle plugin.
+
+If your project is a multi-module project, be sure to read the Working with multi-module projects section on our
+link:maven-tooling#multi-module-maven[Maven] or link:gradle-tooling#multi-module-maven[Gradle] guides.
+
+If you plan to use external modules (for example, an external library for all your domain objects),
+you will need to setup the creation of the index for those modules via the Jandex plugin (if you can modify them)
+or via the `quarkus.index-dependency` inside your `application.properties`.
```

Comment:
Or `META-INF/beans.xml`.

Reply:
I wanted to be more directive than offering the three ways.

The purpose of the guide is to be straightfoward: _if you can add the maven plugin, add it, otherwise use the quarkus.index-dependency properties._

Reply:
Sure. My personal view is `META-INF/beans.xml` is the easiest/quickest way to do that. I am not going to insist here. I did not "request changes" :)

Reply:
I didn't know very well the CDI spec (I'm more used to Spring than J2EE) that why I found strange to the `META-INF/beans.xml`. And there is always the risk that someone finding an empty file on a project just delete it by thinking it was unused ...